internal/common/update/patch_payload.go

```diff
@@ -12,28 +12,49 @@ import (
 )
 
 type attrPatchOperations struct {
-	data          map[string][]jsondiff.Operation
-	ignoreInState []string
+	data                 map[string][]jsondiff.Operation
+	ignoreInStateSuffix  []string
+	ignoreInStatePrefix  []string
+	includeInStateSuffix []string
 }
 
 func (m *attrPatchOperations) ignoreInStatePath(path string) bool {
-	for _, ignore := range m.ignoreInState {
+	for _, include := range m.includeInStateSuffix {
+		suffix := "/" + include
+		if strings.HasSuffix(path, suffix) {
+			return false
+		}
+	}
+	for _, ignore := range m.ignoreInStateSuffix {
 		suffix := "/" + ignore
 		if strings.HasSuffix(path, suffix) {
 			return true
 		}
 	}
+	for _, ignore := range m.ignoreInStatePrefix {
+		for _, part := range strings.Split(path, "/") {
+			if ignore == part {
+				return true
+			}
+		}
+	}
 	return false
 }
 
 func newAttrPatchOperations(patch jsondiff.Patch, options []PatchOptions) *attrPatchOperations {
-	ignoreInState := []string{}
+	ignoreSuffixInState := []string{}
```

Comment:
alternatively you can just initialize them to nil, and append will work fine as well:
```
var (
  ignoreSuffixInState []string
  ignorePrefixInState []string
  includeSuffixInState []string
)
```